mewarpx/mewarpx/diags_store/field_diagnostic.py

```diff
@@ -0,0 +1,93 @@
+from mewarpx.diags_store.diag_base import WarpXDiagnostic
+from mewarpx import plotting
+
+from pywarpx import callbacks, picmi
+from mewarpx import plotting
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+class FieldDiagnostic(WarpXDiagnostic):
+    def __init__(self, diag_steps, diag_data_list, grid, name, write_dir,
+                 plot_on_diag_step=False, plot_data_list=None,**kwargs):
+        """
+        This class is a wrapper for both creating a picmi FieldDiagnostic,
+        and optionally plotting parameters each diagnostic step.
+        Arguments:
+            diag_steps (int): Run the diagnostic with this period.
+                Also plot on this period if enabled.
+            diag_data_list (list (str)): A list of criteria to collect
+                and print diagnostic information for. Accepted values can
+                be found in the picmi standard for field diagnostics.
+            grid (picmi grid): The grid to be passed into the picmi field diagnostic.
+            name (str): The name of the diagnostic to be passed into the
+                picmi field diagnostic.
+            write_dir (str): The directory where diagnostic data will be written
+                from the field diagnostic.
+            plot_on_diag_step (bool): Whether or not to plot data on diagnostic steps.
+            plot_data_list (list (str)): What parameters to plot if plotting on diagnostic steps.
+            kwargs: For a list of valid keyword arguments see diag_base.WarpXDiagnostic
+        """
+        self.diag_steps = diag_steps
+        self.diag_data_list = diag_data_list
+        self.grid = grid
+        self.name = name
+        self.write_dir = write_dir
+        self.plot_on_diag_step = plot_on_diag_step
+        self.plot_data_list = plot_data_list
+
+        super(FieldDiagnostic, self).__init__(diag_steps, **kwargs)
+
+        self.add_field_diag()
+
+        if self.plot_on_diag_step:
+            assert self.plot_data_list is not None, "plot_on_diag_step was True but plot_data_list is None!"
+            self.add_plot_callback()
+
+    def add_field_diag(self):
+        diagnostic = picmi.FieldDiagnostic(
+            grid=self.grid,
+            period=self.diag_steps,
+            data_list=self.diag_data_list,
+            name=self.name,
+            write_dir=self.write_dir
+        )
+
+        mwxrun.simulation.add_diagnostic(diagnostic)
+
+    def add_plot_callback(self):
+        callbacks.installafterstep(self.plot_parameters)
+
+    def plot_parameters(self):
+        if self.check_timestep():
+            current_diag_num = int(mwxrun.get_it() / self.diag_steps)
+            plot_name = "after_diag_step"
+            if current_diag_num is not None:
+                plot_name += "_" + str(current_diag_num)
+
+            for param in self.plot_data_list:
+                if param == 'rho':
+                    data = np.array(mwxrun.get_gathered_rho_grid(include_ghosts=False)[0])
```

Comment:
I think you need to do something like this here since only the root will get the data:
```suggestion
                    data = mwxrun.get_gathered_rho_grid(include_ghosts=False)
                    if mwxrun.me == 0:
                        data = np.array(data[0])
```